### PR TITLE
HPCC-19371 Initial check-in

### PR DESCRIPTION
**Pull request by xwang2713**
This includes Platform Community builds for latest 6.x Gold and rc releases on Ubuntu 16.04 and CentOS 7.
Docker hub HPCCSystems/hpcc project will link master branch and will trigger the build if the project is updated.

There is a build (test-build.sh) script and a test script (test-platform.sh). 
Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com> 

@g-pan, please help to review the README.md here and in https://hub.docker.com/r/hpccsystems/hpcc/ which will automatically linked to this README.md when automated build is done.

@kunalaswani & Michael-Gardner please help to review the code.

Test build (from my branch initial-check-in: https://hub.docker.com/r/hpccsystems/hpcc/builds/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,67 @@
...
+```
+
+To start Docker in daemon mode and map ECLWatch port 8010 to host 8010:
+```sh
+sudo docker run -d --privileged -e "container=docker"-p 8010:8010 hpcc:el7
+```
+
+## How to build Docker HPCC image
+You can always create your own Dockerfile to build a Docker HPCC image. We provide some pre-defined Dockerfiles and tools here.
+These are for HPCC Platform for Ubuntu 16.04 and CentOS 7. HPCC Plugins Dockerfiles to be provided later.
```
Comment (g-pan) [inline README.md]: These are for the HPCC Platform for Ubuntu 16.04 and CentOS 7. HPCC Plugins for Dockerfiles to be provided later.

File: README.md [hunk 2]
```diff
@@ -1 +1,67 @@
...
+
+### Checkout this git reposiotry
+```sh
+git clone https://github.com/hpcc-systems/docker-hpcc.git
+```
+Currently there are two release versions: Gold Release: 6/ and Release Candidate: 6-rc/. There are two product types: platform and plugins. And two Linux distributes: Ubuntu (Ubuntu 16.04, default) and CentOS (CentOS 7).
```
Comment (g-pan) [inline README.md]: plugins. There are two Linux distributions: Ubuntu (Ubuntu 16.04, default) and CentOS (CentOS 7).

File: README.md [hunk 3]
```diff
@@ -1 +1,67 @@
...
+
+### Update HPCC Platform Version
+Make sure the HPCC Platform version is available in HPCCSystems.com -> "Download" -> "HPCC SYSTEM PLATFORM"
+Update the version with update.sh. For example, if you want to build 6.4.14-1:
+```sh
+./update 6.4.14-1
+```
+### Build
+You can go to directory of interest which contains the Dockerfile to run the build, or you can run the test-build.sh script which includes test:
```
Comment (g-pan) [inline README.md]: You can go to the directory of interest which contains the Dockerfile to run the build, or you can run the test-build.sh script which includes test:

**Review by g-pan — COMMENTED** [review]
A few comments in line.

**Comment by xwang2713** [comment]
Updated

**Review by g-pan — APPROVED** [review]

**Comment by xwang2713** [comment]
@richardkchapman I updated the PR and tested locally for HPCC 7.0.0-beta1 builds.  It is ready to merge.

File: 6/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                libblas3              \
+                libatlas3-base        \
+                psmisc                \
+                --fix-missing         \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN groupadd hpcc && useradd -s /bin/bash -r -m -d /home/hpcc -g hpcc -c "hpcc Runtime User" hpcc
+
+ENV VERSION 6.4.16
+ENV FULL_VERSION 6.4.16-1
```
Comment (richardkchapman) [inline 6/Dockerfile:30]: This does not look right...

Comment (xwang2713) [reply 6/Dockerfile:30]: I will update version with 6.4.20-1 is in portal and submit a PR

**Comment by xwang2713** [comment]
Docker build for 7.0 beta1 and 6.4.18-1: https://hub.docker.com/r/hpccsystems/hpcc/tags/
Info: https://hub.docker.com/r/hpccsystems/hpcc/

**Pull request merged**